ee/modules/160-multitenancy-manager/hooks/project.go

```diff
@@ -188,3 +193,65 @@ func readDefaultProjectTemplate(defaultPath, alternativePath string) ([]byte, er
 
 	return projectTemplate, nil
 }
+
+type projectTemplateHelmRenderer struct {
+	projectName string
+}
+
+func (ptr *projectTemplateHelmRenderer) SetProject(name string) {
+	ptr.projectName = name
+}
+
+// Run post renderer which will remove all namespaces except the project one
+// or will add a project namespace if it does not exist in manifests
+func (ptr *projectTemplateHelmRenderer) Run(renderedManifests *bytes.Buffer) (modifiedManifests *bytes.Buffer, err error) {
+	if ptr.projectName == "" {
+		return renderedManifests, nil
+	}
+
+	builder := strings.Builder{}
+
+	manifests := releaseutil.SplitManifests(renderedManifests.String())
+
+	namespaces := make([]v1.Namespace, 0)
+
+	for _, manifest := range manifests {
+		var ns v1.Namespace
+		err = yaml.Unmarshal([]byte(manifest), &ns)
+		if err != nil {
+			return renderedManifests, err
+		}
+
+		if ns.APIVersion != "v1" || ns.Kind != "Namespace" {
+			builder.WriteString("\n---\n" + manifest)
+			continue
+		}
+
+		if ns.Name != ptr.projectName {
+			// drop Namespace from manifests if it's not a project namespace
+			continue
+		}
+
+		namespaces = append(namespaces, ns)
+	}
+
+	result := bytes.NewBuffer(nil)
+
+	for _, ns := range namespaces {
+		if _, ok := ns.GetAnnotations()["multitenancy-boilerplate"]; ok && len(namespaces) > 1 {
+			continue
+		}
+		if ns.Labels == nil {
+			ns.Labels = make(map[string]string, 1)
+		}
+		ns.Labels["d8-module"] = "multitenancy-manager"
```

Comment:
Let's use the `heritage` label. Like `heritage: multitenancy-manager`.
(if I understand correctly for what purpose this label is).

Reply:
Done